core/src/main/java/io/github/fourlastor/game/demo/turns/AttackRanged.java

```diff
@@ -0,0 +1,125 @@
+package io.github.fourlastor.game.demo.turns;
+
+import com.badlogic.gdx.graphics.g2d.TextureAtlas;
+import com.badlogic.gdx.math.MathUtils;
+import com.badlogic.gdx.math.Vector2;
+import com.badlogic.gdx.scenes.scene2d.actions.Actions;
+import com.badlogic.gdx.scenes.scene2d.actions.SequenceAction;
+import com.badlogic.gdx.scenes.scene2d.ui.Image;
+import dagger.assisted.Assisted;
+import dagger.assisted.AssistedFactory;
+import dagger.assisted.AssistedInject;
+import io.github.fourlastor.game.demo.state.GameState;
+import io.github.fourlastor.game.demo.state.unit.Unit;
+
+public class AttackRanged extends TurnState {
+
+    private final StateRouter router;
+
+    private final Unit source;
+    private final Unit target;
+    Vector2 originalPosition = new Vector2();
+    float scale = 1f; // Amount to scale the animation by.
+    int damage = 1;
+    TextureAtlas textureAtlas;
+
+    @AssistedInject
+    public AttackRanged(@Assisted Attack attack, @Assisted TextureAtlas textureAtlas, StateRouter router) {
```

Comment:
if you remove `@Assisted` from `textureAtlas`, it's going to be provided for you automatically (you then can remove it from `Factory` and from `StateRouter`)

`@Assisted` is meant for parameters that are only available at runtime, for example here `Attack` can be anything as it depends on the user choice, so it needs to be `@Assisted`